osquery/remote/http_client.cpp

```diff
@@ -232,7 +231,7 @@ void Client::encryptConnection() {
   ::SSL_set_tlsext_host_name(ssl_sock_->native_handle(),
                              client_options_.remote_hostname_->c_str());
 
-  ssl_sock_->set_verify_callback(boost::asio::ssl::rfc2818_verification(
+  ssl_sock_->set_verify_callback(boost::asio::ssl::host_name_verification(
```

Comment:
[Deprecated and replaced](https://live.boost.org/doc/libs/1_86_0/doc/html/boost_asio/reference/ssl__rfc2818_verification.html)